src/bidiMapper/domains/context/cdpTarget.ts

```diff
@@ -228,6 +230,41 @@ export class CdpTarget {
     }
   }
 
+  // TODO: Document.
+  async addPreloadScriptWithChannels(
+    functionDeclaration: string,
+    channels?: Script.ChannelValue[],
+    sandbox?: string
+  ): Promise<Script.PreloadScript> {
+    // TODO: Document.
+    const createScriptWithChannels = (
+      functionDeclaration: string,
+      channels?: Script.ChannelValue[]
+    ): string => {
+      if (channels === undefined) {
+        // The spec provides a function, and CDP expects an evaluation.
+        return `(${functionDeclaration})();`;
+      }
+
+      const uuid = uuidv4();
+
+      // See: http://go/chrome-devtools:bootstrap-bindings-proposal
+      const intermediateScript = `
+      const channelProxy = (${newChannel()})();
+      window['${uuid}'] = proxy.getMessage;
+      (${functionDeclaration})(...${channels}.map((channel) => channelProxy.getChannel.apply(null, channel)));
```

Comment:
Currently, `channelProxy` does not support multiple channel names, and the channel name is handled by the [long-poll mechanism](https://github.com/GoogleChromeLabs/chromium-bidi/blob/9ce743a425e089d9d7356c8d382a52b594758778/src/bidiMapper/domains/script/scriptEvaluator.ts#L575).

Reply:
Done